### PR TITLE
Test Push API key validation

### PR DESCRIPTION
**Pull request by tombruijn**
Add an extra endpoint to the MockServer to test the Push API key
validation.

I renamed the DiagnoseServer to MockServer to allow it to be used for
more tests.

### CODE REVIEW

**Comment by tombruijn** [comment]
The Elixir and Node.js integrations probably need to be updated to match this before merging.

File: spec/support/server.rb [hunk 1]
```diff
@@ -47,18 +48,35 @@ def last_received_report
       end
     end
 
+    def auth_response_code
+      @mutex.synchronize do
+        @auth_esponse_code
```
Comment (unflxw) [inline spec/support/server.rb:53]:
```suggestion
        @auth_response_code
```

(also in `auth_response_code` and `clear!` below)

File: spec/support/server.rb [hunk 2]
```diff
@@ -47,18 +48,35 @@ def last_received_report
...
 
   post "/diag" do
-    DiagnoseServer.track_request(request)
+    MockServer.track_diagnose_request(request)
     status 200
     JSON.dump({ :token => "diag_support_token" })
   end
+
+  post "/1/auth" do
+    status MockServer.auth_response_code
```
Comment (unflxw) [inline spec/support/server.rb:80]: TODO: check the API key is sent to this endpoint, have an assertion about that? 🤷

File: spec/support/server.rb [hunk 3]
```diff
@@ -21,23 +21,24 @@ def to_h
...
-    def track_request(request)
+    def track_diagnose_request(request)
       @mutex.synchronize do
-        @received_requests << request
+        @received_diagnose_requests << request
       end
     end
 
-    def last_received_report
+    def last_received_diagnose_report
```
Comment (unflxw) [inline spec/support/server.rb:39]: The method name is quite long now, so consider dropping the `received`? (if so, then maybe also from the `@received_diagnose_requests` instance variable)

**Review by unflxw — APPROVED** [review]

**Review by jeffkreeftmeijer — APPROVED** [review]

File: spec/support/server.rb (hunk 1 above)
Comment (tombruijn) [reply spec/support/server.rb:53]: aaaaaaaa I've fixed that three times by now 🤦

File: spec/support/server.rb (hunk 2 above)
Comment (tombruijn) [reply spec/support/server.rb:80]: The difference between 401 and 500 is also still there. And not sure if we want to make that dynamic based on the push_api_key or be more explicit about what status code to return.

**Review by luismiramirez — APPROVED** [review]

**Pull request merged**